packages/backend/src/build-disk-image.ts

```diff
@@ -305,8 +305,27 @@ export async function getUnusedName(name: string): Promise<string> {
 }
 
 // Create builder options for the "bootc-image-builder" container
-export function createBuilderImageOptions(name: string, build: BootcBuildInfo): ContainerCreateOptions {
-  const cmd = [`${build.image}:${build.tag}`, '--output', '/output/', '--local'];
+export async function createBuilderImageOptions(name: string, build: BootcBuildInfo): Promise<ContainerCreateOptions> {
+  // check image for builder to use
+  const image = `${build.image}:${build.tag}`;
+
+  const buildProp = await getConfigurationValue<string>('builder');
+
+  // default to centos builder
+  let builder = bootcImageBuilderCentos;
+  if (buildProp === 'rhel') {
+    // switch to rhel if that's the preference
+    builder = bootcImageBuilderRHEL;
+  } else if (buildProp === 'image') {
+    // or switch to rhel if the preference comes from the image label
+    // AND we detect the rhel label
+    const buildLabel = await containerUtils.getImageBuilderLabel(image);
+    if (buildLabel === 'registry.redhat.io/rhel9/bootc-image-builder') {
+      builder = bootcImageBuilderRHEL;
```

Comment:
this part is confusing to me for the if statement.

If buildProp = rhel, we switch to rhel

i feel like we dont need this check statement as it'll just be a:

if rhel = use this image

if centos or default, use centos

But that's it, no need to rhel check?

Reply:
See https://github.com/CentOS/centos-bootc/pull/453 for some background. The `bootc.diskimage-builder` label allows an image to say 'I expect to be built with builder X'.

If the default preference (image) is used, then we need to look for this label. For now, we only look for one specific value and change to using the RHEL builder. So there are two cases where we'd use the RHEL builder: the preference is explicitly set to prefer RHEL, or preference is image and we detected an image that has asked for this builder.

We could remove the 'image' option and the image label detection for now if you're worried about the cost; users could still switch the builder (for all builds) through the preference.

Reply:
for now I think that we should just have it 100% through preferences.

it's also because I just inspected a few of my images and there is no `bootc.diskimage-builder` for the fedora bootc image...

ex:

```
    "Labels": {
      "containers.bootc": "1",
      "io.buildah.version": "1.35.3",
      "org.opencontainers.image.version": "40.20240503.0",
      "ostree.bootable": "true",
      "ostree.commit": "0b148652f58f77457b63129ae2f95d3f742b521b358727e2daca81fe7cd49e25",
      "ostree.final-diffid": "sha256:12787d84fa137cd5649a9005efe98ec9d05ea46245fdc50aecb7dd007f2035b1",
      "ostree.linux": "6.8.8-300.fc40.x86_64",
      "rpmostree.inputhash": "2d27d73eaa1cf6595b3b5e555ffe0b36a6854740298fa498e8bc1e4872f1d401"
    }
  },
  ```

Reply:
Ok, label detection code removed and simplified just to preferences for now.